crossbeam/datasets/random_data.py

```diff
@@ -230,11 +229,20 @@ def robust_equality(x, y):
 
 def generate_good_random_task(**kwargs):
   """Generates a task that passes simple quality checks."""
+  num_examples = kwargs['num_examples']
+
   while True:
     task = generate_random_task(**kwargs)
     if task is None:
       continue
 
+    # The output can't be a constant across all examples.
+    if num_examples > 1 and task.num_examples == 1:
```

Comment:
do you mind to educate me in what situation would `num_examples != task.num_examples` ?

Reply:
`num_examples` is the number of examples we want to include in the problem.

`task.num_examples` comes from the number of elements in the list of per-example values. However, constants and non-input variables have only 1 element in this list (so we don't have duplication and we don't have to decide the number of examples at the time of creating the constant/variable). This 1 element will get "broadcasted" when needed to compute per-example results -- a value has max(v_i.num_examples) number of examples, over its subtrees v_i. Thus, if the entire expression contains no input variables, the result will only have 1 element in the list.

Reply:
relevant bits of code:
https://github.com/ellisk42/xlambda/pull/1/files/39fc9a435590a4a15258ed241496b8a3bac3015b#diff-16b80d6284a361e58f861620343977db85265c1dcd6be7539333d6367721af5fR65
and
https://github.com/ellisk42/xlambda/pull/1/files/39fc9a435590a4a15258ed241496b8a3bac3015b#diff-9c45164f483ef7eaf7079bd894f11190b1d910dc284fbb3869db563536342571R77

Reply:
Got it, thanks for the explanation!